src/main/java/de/dhbw/mh/rinne/AstBuilder.java

```diff
@@ -61,6 +62,15 @@ public AstNode visitVariableReference(RinneParser.VariableReferenceContext ctx)
     // Team 1
 
     // Team 2
+    @Override
+    public AstNode visitFunctionCall(RinneParser.FunctionCallContext ctx) {
+        CodeLocation codeLoc = getCodeLocation(ctx);
+        String func = ctx.funcCall().getText();
```

Comment:
`ctx.funcCall().getText()` returns the entire raw input matched by `funcCall`, including keywords like `bitte`, `mit`, and the argument list. That’s probably not what you want if you're just trying to extract the function name.

```suggestion
        String functionName = ctx.functionName.getText();
```

This will give you just the identifier bound to `functionName` in your grammar.